groot-btrfs-release/jobs/groot-btrfs/templates/bin/init-store.erb

```diff
@@ -40,15 +40,34 @@ validate_filesystem() {
 
 # Creates a btrfs filesystem and mounts it on $1. The size is defined by $2 in bytes.
 create_and_mount_filesystem() {
-  mkdir -p $1 -m 0755
-  backing_store_file=$1/.backing-store
-  touch $backing_store_file
-  chmod 600 $backing_store_file
-  truncate -s $2 $backing_store_file
-
-  mkfs.btrfs -f $backing_store_file
-  mount -t btrfs -o user_subvol_rm_allowed $backing_store_file $1
-  btrfs quota enable $1
+  mkdir -p "${1}" -m 0755
+  backing_store_file="${1}/.backing-store"
+  touch "${backing_store_file}"
+  chmod 600 "${backing_store_file}"
+  truncate -s "${2}" "${backing_store_file}"
+
+  mkfs.btrfs -f "${backing_store_file}"
+
+  # It is possible for the filesystem creation to take some more time
+  # in the background when reaching the mount, causing it to fail.
+  # Simply repeat the attempt. But not too often, i.e. not too long.
+
+  trials=15
+  sleep_time=2
+  count=$trials
+  until mount -t btrfs -o user_subvol_rm_allowed "${backing_store_file}" "${1}"
+  do
+      count=$(( count - 1 ))
+      if test $count -eq 0 ; then
```

Comment:
[Minor] the reduction of `count` should happen after this check or we only retry 14 times

Reply:
Heh. A bit of interpretation ... 15 trials (as the varname), 14 re-tries. I am letting that one slide.